### PR TITLE
Refactor contact-form

### PR DESCRIPTION
**Pull request by Tishasoumya-02**
#4460

### CODE REVIEW

**Comment by avoinea** [comment]
@Tishasoumya-02 Can you please add a reference to the initial github issue for this PR?!

**Comment by Tishasoumya-02** [comment]
Okay Ya sure @avoinea

File: cypress/tests/core/basic/contactForm.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+describe('Contact Form Tests', () => {
+    beforeEach(() => {
+      cy.visit('/');
+    });
+    it('Control Form', function () {
```
Comment (nileshgulia1) [inline cypress/tests/core/basic/contactForm.js:5]:
```suggestion
    it('renders contact Form', function () {
```

File: cypress/tests/core/basic/contactForm.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      cy.get('a:contains("Contact")').click();
+      cy.get('#fieldset-undefined-field-label-name').type('Myname');
```
Comment (nileshgulia1) [inline cypress/tests/core/basic/contactForm.js:7]: Please check why there are selectors with undefined names?

**Review by nileshgulia1 — CHANGES_REQUESTED** [review]

File: src/hooks/emailNotification/useEmailNotification.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { useSelector } from 'react-redux';
+
+/**
+ * useEmailNotification hook
+ *
+ * This hook returns the current status of email Notification that is stored in the Redux store in the
```
Comment (nileshgulia1) [inline src/hooks/emailNotification/useEmailNotification.js:6]: @Tishasoumya-02 Also please fix the grammer! :)

Comment (Tishasoumya-02) [reply src/hooks/emailNotification/useEmailNotification.js:6]: yup my mistake

File: src/components/theme/ContactForm/ContactForm.jsx [hunk 4]
```diff
@@ -65,195 +59,126 @@ const messages = defineMessages({
...
+  }),
+  loading: PropTypes.bool,
+  loaded: PropTypes.bool,
+};
+
+ContactFormComponent.defaultProps = {
+  error: null,
+  loading: null,
+  loaded: null,
+};
```
Comment (nileshgulia1) [inline src/components/theme/ContactForm/ContactForm.jsx:182]: These are now coming via hooks. You can remove them.

**Review by nileshgulia1 — CHANGES_REQUESTED** [review]

**Review by nileshgulia1 — APPROVED** [review]

**Review by nileshgulia1 — APPROVED** [review]

**Pull request closed without merging**